packages/toolpad-app/src/toolpad/propertyControls/SelectOptions.tsx

```diff
@@ -72,14 +80,20 @@ function SelectOptionsPropEditor({
 
   const handleOptionItemClick = React.useCallback(
     (index: number) => () => {
+      setOptionTextInputInvalid(false);
       setEditingIndex(index);
     },
     [],
   );
 
   const handleOptionChange = React.useCallback(
     (newValue: SelectOption) => {
-      onChange(value.map((option, i) => (i === editingIndex ? newValue : option)));
+      let newOption: string | SelectOption = newValue;
```

Comment:
`newValue.value` sounds a bit confusing to read - what about calling it `newOption` instead?

Reply:
Makes sense, I can actually get rid of that variable itself and just use type narrowing